### PR TITLE
server,eth: add method for account balance retrieval

### PR DESCRIPTION
**Pull request by buck54321**
The interfaces are going to need to diverge when it comes to tracking funding ability. This PR is a first thrust at a reasonable interface scheme so that we can begin work in `Swapper`, `Market`, and `CoinLocker`.

### CODE REVIEW

**Review by JoeGruffins — APPROVED** [review]
Looks reasonable to me. Please add meaningful commit messages.

**Comment by chappjc** [comment]
Needs resolve after txData PR merged.  Might as well squash down with a final commit too.

File: server/asset/eth/rpcclient.go [hunk 1]
```diff
@@ -120,3 +120,13 @@ func (c *rpcclient) swap(ctx context.Context, secretHash [32]byte) (*swap.ETHSwa
 func (c *rpcclient) transaction(ctx context.Context, hash common.Hash) (tx *types.Transaction, isMempool bool, err error) {
 	return c.ec.TransactionByHash(ctx, hash)
 }
+
+// accountBalance gets the account balance, including the effects of known
+// unmined transactions.
+func (c *rpcclient) accountBalance(ctx context.Context, addr common.Address) (*big.Int, error) {
+	bigBal, err := c.ec.PendingBalanceAt(ctx, addr)
```
Comment (chappjc) [inline server/asset/eth/rpcclient.go:127]: I wonder how this works with multiple txns with the same nonce, e.g. replacements commonly used to either cancel a txn or speed up a txn

Comment (chappjc) [reply server/asset/eth/rpcclient.go:127]: Exactly, a 0 ETH transaction to yourself.  At least that's how I've seen it on etherscan.

**Review by chappjc — APPROVED** [review]

**Pull request merged**